### PR TITLE
blocky: embed version info

### PR DESCRIPTION
**Pull request by bjornfor**
###### Description of changes
Before:

```
  $ blocky version
  [...]
  Version: undefined
```

After:

```
  $ blocky version
  [...]
  Version: 0.19
```

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by ratsclub — APPROVED** [review]

File: pkgs/applications/networking/blocky/default.nix [hunk 1]
```diff
@@ -21,6 +21,8 @@ buildGoModule rec {
 
   vendorSha256 = "sha256-fsMBL9qyhIrV6eAsqpSaNniibMdYRVBnl2KCzStvMGQ=";
 
+  ldflags = [ "-X github.com/0xERR0R/blocky/util.Version=${version}" ];
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/blocky/default.nix:24]:
```suggestion
  ldflags = [ "-s" "-w" "-X github.com/0xERR0R/blocky/util.Version=${version}" ];
```
lets also tell go to strip

Comment (bjornfor) [reply pkgs/applications/networking/blocky/default.nix:24]: @SuperSandro2000: Thanks, I didn't know. (I couldn't find `-w` option in `man ld`, but apparently it disables DWARF debug info.) But I feel that's off topic for "embed version info". Longer term, maybe `ldflags = ["-s" "-w"]` is something buildGoModule should default to?

Comment (SuperSandro2000) [reply pkgs/applications/networking/blocky/default.nix:24]: https://github.com/NixOS/nixpkgs/issues/177698

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Pull request merged**